src/config.rs

```diff
@@ -334,6 +334,90 @@ impl Context {
     }
 }
 
+// Separate impl block for self address handling
+impl Context {
+    /// determine whether the specified addr maps to the/a self addr
+    pub(crate) async fn is_self_addr(&self, addr: &str) -> Result<bool> {
+        let addr = addr_normalize(addr).to_lowercase();
+
+        // The addresses we get here are already normalized and lowercase
+        Ok(
+            self.get_primary_self_addr().await.ok().as_deref() == Some(&addr)
+                || self.get_secondary_self_addrs().await?.contains(&addr),
+        )
+    }
+
+    /// Sets `primary_new` as the new primary self address and saves the old
+    /// primary address (if exists) as a secondary address.
+    pub(crate) async fn set_primary_self_addr(&self, primary_new: &str) -> Result<()> {
+        let primary_new = addr_normalize(primary_new).to_lowercase();
+
+        if self.get_primary_self_addr().await.ok().as_deref() != Some(&primary_new) {
+            // Make sure that all server UIDs will be resynced
```

Comment:
Is clearing `imap_sync` the right way to do it? Why not call `job::schedule_resync`?

Actually I don't get why this is done here. Seems error-prone that such IMAP-specific changes go into identity-management code. If UID validity is not enough to detect account change, maybe `imap_sync` should have a column specifying which account this data is valid for, so IMAP code can treat account change the same way as it treats UID validity change?

Reply:
> Is clearing `imap_sync` the right way to do it? Why not call `job::schedule_resync`?

Probably makes more sense (I had feared that the resync may be "too late" but since jobs are executed before fetching, it won't come not too late).

> If UID validity is not enough to detect account change ...

I don't know if it's enough; on my GMX accounts it would have been enough since UID validity is a very large number, but I didn't want to rely on this.

> …maybe `imap_sync` should have a column specifying which account this data is valid for, so IMAP code can treat account change the same way as it treats UID validity change?

Sounds like more effort and more lines added, but if you say you think it's worth it, I'll do it.
Another option would be to move the call to `configure` where it seems less out of place than in an identity management function.

Reply:
Just a `schedule_resync` is probably ok, we can return to it later as we keep removing jobs.